tests/models/kosmos2_5/test_modeling_kosmos2_5.py

```diff
@@ -0,0 +1,611 @@
+        inputs = {k: v.to(torch_device) if v is not None else None for k, v in inputs.items()}
+        inputs["flattened_patches"] = inputs["flattened_patches"].to(model.dtype)
+
+        generation_outputs = model.generate(
+            **inputs,
+            max_new_tokens=1024,
+        )
+        generated_ids = generation_outputs
+        generated_text = processor.batch_decode(generated_ids, skip_special_tokens=True)
+
+        return generated_ids, generated_text
+
+    def test_receipt_image_ocr(self):
+        url = "https://huggingface.co/microsoft/kosmos-2.5/resolve/main/receipt_00008.png"
+        url = "https://huggingface.co/kirp/kosmos2_5/resolve/main/receipt_00008.png"
+        image = Image.open(requests.get(url, stream=True).raw)
+
+        dtype = torch.bfloat16
+        repo = "microsoft/kosmos-2.5"
+        model = Kosmos2_5ForConditionalGeneration.from_pretrained(repo, device_map=torch_device, torch_dtype=dtype)
+        processor = AutoProcessor.from_pretrained(repo)
+        prompt = "<ocr>"
+        generated_ids, generated_text = self.run_example(prompt, image, model, processor)
+
+        EXPECTED_TEXT = [
+            "<ocr><bbox><x_53><y_573><x_69><y_606></bbox>1\n<bbox><x_79><y_573><x_464><y_611></bbox>[REG] BLACK SAKURA\n<bbox><x_690><y_569><x_810><y_606></bbox>45,455\n<bbox><x_53><y_614><x_69><y_648></bbox>1\n<bbox><x_79><y_614><x_468><y_650></bbox>COOKIE DOH SAUCES\n<bbox><x_788><y_609><x_812><y_644></bbox>0\n<bbox><x_50><y_658><x_69><y_693></bbox>1\n<bbox><x_79><y_658><x_358><y_693></bbox>NATA DE COCO\n<bbox><x_790><y_652><x_814><y_687></bbox>0\n<bbox><x_31><y_742><x_820><y_781></bbox>Sub Total 45,455\n<bbox><x_27><y_781><x_822><y_827></bbox>PB1 (10%) 4,545\n<bbox><x_27><y_826><x_824><y_872></bbox>Rounding 0\n<bbox><x_24><y_872><x_827><y_921></bbox>Total 50,000\n<bbox><x_17><y_1056><x_836><y_1108></bbox>Card Payment 50,000\n"
+        ]
+
+        self.assertListEqual(generated_text, EXPECTED_TEXT)
+
+    def test_receipt_image_md(self):
+        url = "https://huggingface.co/microsoft/kosmos-2.5/resolve/main/receipt_00008.png"
+        url = "https://huggingface.co/kirp/kosmos2_5/resolve/main/receipt_00008.png"
+        image = Image.open(requests.get(url, stream=True).raw)
+
+        dtype = torch.bfloat16
+        repo = "microsoft/kosmos-2.5"
+        model = Kosmos2_5ForConditionalGeneration.from_pretrained(repo, device_map=torch_device, torch_dtype=dtype)
+        processor = AutoProcessor.from_pretrained(repo)
+        prompt = "<md>"
+        generated_ids, generated_text = self.run_example(prompt, image, model, processor)
+        print(generated_text)
+        EXPECTED_TEXT = [
+            "<md>- **1 \\[REG\\] BLACK SAKURA** 45,455\n- **1 COOKIE DOH SAUCES** 0\n- **1 NATA DE COCO** 0\n- **Sub Total** 45,455\n- **PB1 (10%)** 4,545\n- **Rounding** 0\n- **Total** **50,000**\n\nCard Payment 50,000"
+        ]
+        self.assertListEqual(generated_text, EXPECTED_TEXT)
```

Comment:
added

Reply:
Thank you. Since our CI is still using T4 runner, and non of these 3 tests pass (GPU OOM), I am thinking to reduce `max_new_tokens=1024,` to something smaller. 

Do you have any comment about this?

Reply:
If you think that makes sense too, I can run on T4 and update the expected output values on my own side

Reply:
well, it turns out that GPU OOM happens already at `vision_model_output = self.vision_model`. So reducing `max_new_tokens` won't work here. Forget about my above comment.